### PR TITLE
Adding boss cores to boss inventory

### PR DESCRIPTION
**Pull request by MasterGordon**

### CODE REVIEW

File: game/scripts/vscripts/components/boss/spawn.lua [hunk 1]
```diff
@@ -107,6 +107,18 @@ function BossSpawner:SpawnBoss (pit, boss, bossTier, isProtected)
 
   bossHandle:AddItem(heart)
 
+  local core = CreateItem("item_upgrade_core", bossHandle, bossHandle)
```
Comment (chrisinajar) [inline game/scripts/vscripts/components/boss/spawn.lua]: do your conditional chains like...
```lua
local core = nil

if bossTier < 1 then
  core = CreateItem("item_upgrade_core", bossHandle, bossHandle)
elseif bossTier == 1 then
  core = CreateItem("item_upgrade_core_2", bossHandle, bossHandle)
elseif bossTier == 2 then
  core = CreateItem("item_upgrade_core_3", bossHandle, bossHandle)
elseif bossTier >= 3 then
  core = CreateItem("item_upgrade_core_4", bossHandle, bossHandle)
end

if core == nil then
  error('Got bad core, tier must have bad value ' .. tostring(bossTier))
end
```

**Comment by MasterGordon** [comment]
Thanks for the feedback.

**Comment by imaGecko** [comment]
did you test how this interacts with uriel?

**Comment by MasterGordon** [comment]
Worked fine with uriel he just replaces the core when he uses his spell

**Comment by chrisinajar** [comment]
Nice work, this will make things more clear to players :+1:

**Pull request merged**